src/test/java/seedu/address/model/delivery/DateTimeTest.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.delivery;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DateTimeTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DateTime(null));
+    }
+
+    @Test
+    public void constructor_invalidTime_throwsIllegalArgumentException() {
+        String invalidTime = "01-01-2234 00:70";
+        assertThrows(IllegalArgumentException.class, () -> new DateTime(invalidTime));
+    }
+
+    @Test
+    public void isValidDateTime() {
+        // null time
+        assertThrows(NullPointerException.class, () -> DateTime.isValidTime(null));
+
+        // invalid time formats
+        assertFalse(DateTime.isValidTime("")); // empty string
+        assertFalse(DateTime.isValidTime(" ")); // spaces only
+        assertFalse(DateTime.isValidTime("10-10-2024 50:00")); // invalid date and time
```

Comment:
Perhaps we can separate this to test for invalid date and invalid time separately?